### PR TITLE
Feature: let dledger complete responseFutures after applying entries to statemachine

### PR DESCRIPTION
**Pull request by hzh0425**
let dledger complete responseFutures after applying entries to statemachine

### CODE REVIEW

**Comment by hzh0425** [comment]
@RongtongJin please take a look

File: src/main/java/io/openmessaging/storage/dledger/DLedgerEntryPusher.java [hunk 1]
```diff
@@ -243,73 +315,69 @@ public void doWork() {
...
-                            response.setTerm(currTerm);
-                            response.setIndex(i);
-                            response.setLeaderId(memberState.getSelfId());
-                            response.setPos(((AppendFuture) future).getPos());
-                            future.complete(response);
-                        }
-                        ackNum++;
-                    } catch (Throwable t) {
-                        logger.error("Error in ack to index={} term={}", i, currTerm, t);
+                        cb.get(10, TimeUnit.SECONDS);
```
Comment (RongtongJin) [inline src/main/java/io/openmessaging/storage/dledger/DLedgerEntryPusher.java:334]: 为什么需要一个CompletableFuture在这里做等待吗？因为这并不是返回给客户端的CompletableFuture

File: src/main/java/io/openmessaging/storage/dledger/statemachine/CommittedEntryIterator.java [hunk 2]
```diff
@@ -56,7 +60,17 @@ public DLedgerEntry next() {
         return null;
     }
 
+    public void completeApplyingEntry() {
+        if (this.completeEntryCallback.apply(this.currentIndex)) {
+            this.completeAckNums++;
+        }
+    }
```
Comment (RongtongJin) [inline src/main/java/io/openmessaging/storage/dledger/statemachine/CommittedEntryIterator.java:67]: 需要用户自己再每一个apply后调用吗？

File: src/main/java/io/openmessaging/storage/dledger/DLedgerEntryPusher.java (hunk 1 above)
Comment (hzh0425) [reply src/main/java/io/openmessaging/storage/dledger/DLedgerEntryPusher.java:334]: 因为要等 apply 完

File: src/main/java/io/openmessaging/storage/dledger/statemachine/CommittedEntryIterator.java (hunk 2 above)
Comment (hzh0425) [reply src/main/java/io/openmessaging/storage/dledger/statemachine/CommittedEntryIterator.java:67]: 是的, apply 完后调用, 然后回调通知.

**Review by RongtongJin — COMMENTED** [review]
It would be better to print applied index watermarks regularly like committed index

**Comment by RongtongJin** [comment]
LGTM～

**Pull request merged**